dali/python/nvidia/dali/ops/_signatures.py

```diff
@@ -0,0 +1,393 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from inspect import Parameter, Signature
+import ast
+import os
+
+from pathlib import Path
+
+from typing import Union, Optional
+from typing import Sequence, Any
+
+from nvidia.dali import backend as _b
+from nvidia.dali import types as _types
+from nvidia.dali.ops import _registry, _names, _docs
+from nvidia.dali.fn import _to_snake_case
+from nvidia.dali import types
+
+
+def _create_annotation_placeholder(typename):
+    # Inspect and typing know better how to format annotations. They allow of short versions
+    # of anything from typing module and for builtins, but anything else gets a fully qualified
+    # path and there is no customization. Also the formatting using the `repr` of the type is
+    # driven by inspect.formatannotation (when we use the type directly) or by the internals of
+    # typing classes like Union. So we just pretend, that we are from typing, and we want to
+    # have our name as `typename`.
+
+    class _AnnotationPlaceholderMeta(type):
+        # We don't want <class 'typename'> when inspect calls inspect.formatannotation directly.
+        def __repr__(cls):
+            return typename
+
+    class _AnnotationPlaceholder(metaclass=_AnnotationPlaceholderMeta):
+        # typing uses those instead of repr
+        __name__ = typename
+        __qualname__ = typename
+        # only if you are part of `typing.` your path is hidden
+        __module__ = "typing"
+
+        def __init__(self, val):
+            self._val = val
+
+        # Work around the repr of enums, by replacing them with `__str__`
+        def __repr__(self):
+            return str(self._val)
+
+    return _AnnotationPlaceholder
+
+
+# This is not the DataNode you are looking for.
+_DataNode = _create_annotation_placeholder("DataNode")
+
+# The placeholder for the DALI Enum types, as the bindings from backend don't play nice,
+# we need actual Python classes.
+_DALIDataType = _create_annotation_placeholder("DALIDataType")
+_DALIImageType = _create_annotation_placeholder("DALIImageType")
+_DALIInterpType = _create_annotation_placeholder("DALIInterpType")
+
+_enum_mapping = {
+    types.DALIDataType: _DALIDataType,
+    types.DALIImageType: _DALIImageType,
+    types.DALIInterpType: _DALIInterpType
+}
+
+_MAX_INPUT_SPELLED_OUT = 5
+
+
+# TODO(klecki): Propagate type hints for those functions
+# Those operators have custom Python code, so the annotations are done in-place for the
+# wrappers rather than via automatic generation.
+_manual_definitions = [
+    "ExternalSource",
+    "TFRecordReader",
+    "readers__TFRecord",
+    "PythonFunction",
+    "DLTensorPythonFunction",
+    "NumbaFunction",
+]
```

Comment:
Don't we have some other way to detect those? I think there's some mechanism in the (former) ops.py that generated (or not) those functions.
How about adding a field `generated` or similar and only processing those here?